modules/renter/renter.go

```diff
@@ -193,18 +193,18 @@ type Renter struct {
 	lastEstimation modules.RenterPriceEstimation
 
 	// Utilities.
-	chunkCache     map[string]*cacheData
-	cmu            *sync.Mutex
-	cs             modules.ConsensusSet
-	deps           modules.Dependencies
-	g              modules.Gateway
-	hostContractor hostContractor
-	hostDB         hostDB
-	log            *persist.Logger
-	persistDir     string
-	mu             *siasync.RWMutex
-	tg             threadgroup.ThreadGroup
-	tpool          modules.TransactionPool
+	staticStreamCache *streamCache
```

Comment:
does it make sense to call this `static`? I guess the renter doesn't modify it directly, but it's still modified by its methods, same as `tpool`/`hostDB`. I would just call it `streamCache`.

Reply:
I think it makes sense since `static` refers to the lock of the object that holds the `static` field. That means I only have read-only access to the pointer and can't replace it during the lifetime of the renter.
When I call a private method on a `static` object I still need to hold the object's lock, but I know that I don't have to hold the Renter's lock. On the other hand if I call a public method on the `static` object I can't hold any locks even though `static` usually means I can hold the lock of the owning object if I want to.

Reply:
That's btw also the reason why the `staticContracts` in the contractor are static now, even though the field has methods that modify its contents.

Reply:
Doesn't the `streamCache` handle all of its own locking internally though? Speaking of which, do we still need `cmu`?

There isn't really a precedent for this so we'll have to use our best judgment. The `streamCache` looks to me like a dependency, not very different from the `tpool` or `hostDB`. If it could (in theory) be moved into its own package, I think that qualifies it as a dependency, and we currently don't apply the `static` convention to dependencies.

Reply:
We can probably get rid of the `cmu`.

Well, if we consider the `contractSet` a dependency, then we do apply the `static` convention already. The problem is, that there is no way to tell if you have to hold the lock of the parent object without making it `static`. In my opinion that introduces unnecessary ambiguity when mixing dependencies and simple helper structs.
I also don't think that calling a method that modifies the object's fields on an object violates the `static` rule. Static refers to the value of the pointer, which doesn't change since it is only a reference.

Reply:
good point about the contract set. So there is a precedent, but we aren't very consistent about it; `hostContractor` stores the field as `staticContracts`, but `proto.Editor` and `proto.Downloader` just call it `contractSet`.

I think I see what you're getting at. When we call a method on a dependency, e.g. `r.tpool.AcceptTransactionSet`, we assume that `tpool` will handle the locking. But for a "helper" object, like `file`, it may be necessary to hold the parent lock. So how do you tell when you need to hold the lock?

Things get messy because sometimes you want to hold the parent lock throughout multiple helper operations, like calling multiple methods on a `file`. This is (IIRC) disallowed for dependencies because of deadlock risk. I think we previously encountered this when splitting out the `hostTree` from the `hostDB`.

Worse, I don't think we can introduce a convention that alters method names (like `managed`) because that could prevent a helper type from satisfying an interface. For example, you can't rename `streamHeap.Pop` to `streamHeap.managedPop`.

So I'm not really sure how to handle these helper objects. I don't think it's *wrong* to call them static, because it's true that the parent doesn't directly modify the pointer itself. But I also feel like it's potentially confusing, because the nature of the helper object isn't "static" at all; it has lots of internal state that changes frequently.

Reply:
Yes, exactly. It's probably not a problem for 'real' dependencies from different packages since you can only access them via exported methods anyway (and you are not allowed to use those while holding a lock). 
Otherwise I think calling it `static` is a good enough solution for now.